### DISCUSSION TITLE
How to know if SubGen works

### DISCUSSION THREAD

**Discussion by Naldjer** [McCloudS/subgen#155] [Q&A] [answered]
hello,

Im running SubGen in Truenas as an app, imported by the .Yaml file.
I also got Bazarr running with WhisperAI (pointing to SubGen) as provider.
Configured the webhook in Plex, but how can i see that SubGen creates subtitles? The CPU of the application is 0 all the time, and the logs arent providing me any (usefull) information. Just numbers. 

Plexpath = 
   /mnt/Plex/Movies
   /mnt/Plex/TVShows
SubGen paths:
   /mnt/Plex/Movies
   /mnt/Plex/TVShows

URL configured in Plex as Webhook:
IP SubGen:9000/plex

Many thanks!

Kind regards,

Naldjer

**Comment by McCloudS** [comment]
What are your config values?  Did you set your plex token?  Debug to true will give more info, which would help figure out your problem.

**Comment by Naldjer** [comment]
Hi friend,

Debug in the config files has already been set to true.
Hereby the config file:

services:
  subgen:
    container_name: subgen
    environment:
      - WHISPER_MODEL=medium
      - WHISPER_THREADS=4
      - PROCADDEDMEDIA=True
      - PROCMEDIAONPLAY=False
      - NAMESUBLANG=aa
      - SKIPIFINTERNALSUBLANG=eng
      - PLEXTOKEN={Plex token which i retrieved from the Get-Info > View XML step}
      - PLEXSERVER=http://192.168.178.25:32400
      - JELLYFINTOKEN=token here
      - JELLYFINSERVER=http://jellyfin:8096
      - WEBHOOKPORT=9000
      - CONCURRENT_TRANSCRIPTIONS=2
      - WORD_LEVEL_HIGHLIGHT=False
      - DEBUG=True
      - USE_PATH_MAPPING=False
      - PATH_MAPPING_FROM=/tv
      - PATH_MAPPING_TO=/Volumes/TV
      - TRANSCRIBE_DEVICE=cpu
      - CLEAR_VRAM_ON_COMPLETE=True
      - MODEL_PATH=./models
      - UPDATE=False
      - APPEND=False
      - USE_MODEL_PROMPT=False
      - CUSTOM_MODEL_PROMPT=
      - LRC_FOR_AUDIO_FILES=True
      - CUSTOM_REGROUP=cm_sl=84_sl=42++++++1
    image: mccloud/subgen
    ports:
      - '9000:9000'
    tty: True
    volumes:
      - /mnt/Plex/TVShows
      - /mnt/Plex/Movies
      - /mnt/Plex/Subs
version: '2'

Logs right now:
![image](https://github.com/user-attachments/assets/2ea18205-4900-4306-b6af-1e479af24e66)

**Comment by McCloudS** [comment]
Your picture doesn’t really show anything, but I would guess you want PROCMEDIAONPLAY True, otherwise subgen would only run when a new file is added to a plex library.

**Comment by Naldjer** [comment]
Hi,

even when there has been a new file added on Plex, the CPU load of SubGen keeps 0%. So i dont have any clue if SubGen is working or not...

**Reply by McCloudS** [reply]
What do the logs show?  If plex is sending events correctly, subgen will see them.

**Reply by Naldjer** [reply]
> What do the logs show? If plex is sending events correctly, subgen will see them.

If the logs are stored elsewhere i would love to know where. When i view the logs from out of Docker (or kubernetes, i dont know what Truenas uses) i get the same output as above

**Reply by McCloudS** [reply]
THey are output to the docker logs/console for the container, however you view that.  The same as any other container.  

You should be seeing something similar to on startup: 
![image](https://github.com/user-attachments/assets/7563b152-f5d0-480b-b731-3c3df71559fe)

**Reply by Naldjer** [reply]
I see this: 
![image](https://github.com/user-attachments/assets/a6818347-959b-45be-9a11-9e157692f89b)

**Reply by McCloudS** [reply]
Maybe your container is corrupted?  The dates don't line up and are progressing in odd ways.  I'm not sure of your configuration, so I'm not able to help with that.

**Reply by Naldjer** [reply]
Idk if it helps anything, but this is what Bazarr say:

![image](https://github.com/user-attachments/assets/56fd2d73-cb59-44ac-b14a-cb4fca284f09)

**Reply by McCloudS** [reply]
At least that shows the container is running and responding, but whatever reason isn’t logging to your console correctly.  Not sure what to do there.

**Reply by Naldjer** [reply]
Okay, well ill continue puzzling a bit. Many thanks for your help, and this amazing application :) I really hope ill get it to work :D

**Reply by Naldjer** [reply]
Ok so instead of importing it via yml, ive configured it manually. The logs now look like this:

![image](https://github.com/user-attachments/assets/0ac7f7b7-fff5-43e0-b176-dceded389033)

**Comment by McCloudS** [comment, marked as answer]
Great.  Now if you try adding a video, it should populate something in the
log.  It’s easier to troubleshoot with onplay and just play videos.

**Reply by Naldjer** [reply]
Okay, i will enable the onplay setting and come back right after :)

**Reply by Naldjer** [reply]
Looks good, innit?

![image](https://github.com/user-attachments/assets/1888025e-f6e8-4fc8-a408-4f9ca34bae10)

Looks like only Bazarr now is not using SubGen for subtitle generation

**Reply by McCloudS** [reply]
Yup, you should expect that.  Looks like it didn’t generate anything because it already has a internal that matches your requirements.

**Reply by Naldjer** [reply]
I have like 300+ missing subtitles in Bazarr for different movies / tvshows. But Bazarr doesnt seem to put SubGen to work. Is that by default, or do i have to enable something for that?

**Reply by McCloudS** [reply]
> I have like 300+ missing subtitles in Bazarr for different movies / tvshows. But Bazarr doesnt seem to put SubGen to work. Is that by default, or do i have to enable something for that?

You’ll have to get support in the whisper channel on discord.  Likely your score is too low for whisper to run.  Needs to be under 65 or so.

**Reply by Naldjer** [reply]
Okay, i will definiatly do that. Many thanks for your amazing support! Im so happy it works :D . One more question : If enabled PROCONPLAY - does the subtitles automatically appear in the movie, or are they available after the movie has played?

**Reply by Naldjer** [reply]
Do you also have got the discord link of the server you mentioned by any chance? Tried to Google it but cant find it

**Reply by McCloudS** [reply]
You’ll have to select them in plex after they are generated.  If you’re on a CPU, they likely wouldn’t be done by the time you finished watching a file.  Onplay works well for my gpu, where I can start it, pause it for a minute or two, then enable the subtitles.

**Reply by McCloudS** [reply]
> Do you also have got the discord link of the server you mentioned by any chance? Tried to Google it but cant find it

https://discord.gg/MH2e2eb

**Reply by Naldjer** [reply]
No wonder i couldnt find it,,, i was looking through SubGen git and after that the Whisper AI git lol,,,

anyway,, i got a file playing now. This is the log:

![image](https://github.com/user-attachments/assets/8a75b1e0-dcbb-4713-b530-be1178c1b6e2)


Yet, the CPU load is 0%:

![image](https://github.com/user-attachments/assets/bcd737c3-cffa-4078-a2bf-c5efe535d553)

Is that normal behaviour?

**Reply by McCloudS** [reply]
Yes.  It’s skipping generation because you have SKIPIFINTERNALSUBLANG

**Reply by Naldjer** [reply]
I see it, but only when it has a ENG language. The internal language in this videofile is Chinese

![image](https://github.com/user-attachments/assets/bf6b925d-6bcf-4cc7-b099-78c3d7baf2de)

I think i posted the wrong log:

![image](https://github.com/user-attachments/assets/612fd15f-ea1f-4fc7-bd78-3d915c230adc)

**Reply by McCloudS** [reply]
The raw plex responses look fine, but there should be more logging after a play event (if you still have on play enabled).

**Reply by Naldjer** [reply]
Yep, havent turned it off

**Reply by Naldjer** [reply]
Got some more information now:

![image](https://github.com/user-attachments/assets/ff085cac-a5ad-4705-bce9-83dcc706bc78)

**Reply by McCloudS** [reply]
WHISPER_MODEL 4 isn’t a valid option.

**Reply by Naldjer** [reply]
Ah i see, changed it to medium now as that is the default value:

![image](https://github.com/user-attachments/assets/5d30beb2-002f-417d-bee5-8fc8ff75a2f4)

**Reply by McCloudS** [reply]
Looks like your container doesn't have access to the internet (or huggingfacehub) to download the models.

**Reply by Naldjer** [reply]
Would that cause troubles? Cuz it seems to work now:

![image](https://github.com/user-attachments/assets/34e17055-85f0-4bd4-ae82-db54c1bb66f4)

**Reply by McCloudS** [reply]
It needs to be able to download the whisper model at least once or it has nothing to run.  Maybe it finally downloaded.  Guess you can wait and see if you start seeing progress.

**Reply by Naldjer** [reply]
Would it be better to instal SubGen on my laptop which has this GPU? The CPU in the other machine is a Intel Xeon 3340 6 core

![image](https://github.com/user-attachments/assets/35bff321-32ac-46ff-b645-981e6da19334)

**Reply by McCloudS** [reply]
If it works, it will be better.  I’m unsure if the MX250 would even be supported.

**Reply by Naldjer** [reply]
Okay, im installing Ubuntu now on a VM. Keep you updated. :) Many thanks for now!

**Reply by Naldjer** [reply]
With that Xeon processor i could watch the movie 10 times before the subs are ready LOL

**Reply by Naldjer** [reply]
Okay, so the first subs should be ready. But ended in an error:

![image](https://github.com/user-attachments/assets/f8405e85-d0c6-4b9e-bf21-3cbf9f929146)

**Reply by McCloudS** [reply]
You hand typed the regroup env variable wrong.

**Reply by Naldjer** [reply]
Yea figured it out. Changed s1 to sl. retrying now :)

**Reply by Naldjer** [reply]
Ello,

The application working fine so far. Had some troubles with the manual typing of the regroup var but that has been fixed now.
I have one more question tho. Is it also possible to get subs in other languages such as Dutch subs for instance?

Many thanks!

**Reply by Naldjer** [reply]
Hey,

On Windows, do i have to use USE_PATH_MAPPING or where do i have to define the pathstructure?

You need to have matching paths relative to your Plex server/folders, or use USE_PATH_MAPPING. Paths are not needed if you are only using Bazarr
